example/command_runner/draw.dart

```diff
@@ -0,0 +1,141 @@
+      final tx = 2 * x / width - 1;
+      out.write(pixel(tx, ty) ? char : ' ');
+    }
+    out.write('\n');
+  }
+  return out.toString();
+}
+
+String drawTriangle(int width, int height, String char) {
+  return draw(width, height, char, (x, y) => x.abs() <= (1 + y) / 2);
+}
+
+class SquareCommand extends Command<String> {
+  SquareCommand() {
+    argParser.addOption('size', help: 'Size of the square');
+  }
+
+  @override
+  String get name => 'square';
+
+  @override
+  String get description => 'Draws a square';
+
+  @override
+  List<String> get aliases => ['s'];
+
+  @override
+  FutureOr<String>? run() {
+    final size = int.parse(argResults?['size'] ?? '20');
+    final char = globalResults?['char']?[0] ?? '#';
+    return draw(size, size, char, (x, y) => true);
+  }
+}
+
+class CircleCommand extends Command<String> {
+  CircleCommand() {
+    argParser.addOption('radius', help: 'Radius of the circle');
+  }
+
+  @override
+  String get name => 'circle';
+
+  @override
+  String get description => 'Draws a circle';
+
+  @override
+  List<String> get aliases => ['c'];
+
+  @override
+  FutureOr<String>? run() {
+    final size = 2 * int.parse(argResults?['radius'] ?? '10');
+    final char = globalResults?['char']?[0] ?? '#';
+    return draw(size, size, char, (x, y) => x * x + y * y < 1);
+  }
+}
+
+class TriangleCommand extends Command<String> {
+  TriangleCommand() {
+    addSubcommand(EquilateralTriangleCommand());
+    addSubcommand(IsoscelesTriangleCommand());
+  }
+
+  @override
+  String get name => 'triangle';
+
+  @override
+  String get description => 'Draws a triangle';
+
+  @override
+  List<String> get aliases => ['t'];
+}
+
+class EquilateralTriangleCommand extends Command<String> {
+  EquilateralTriangleCommand() {
+    argParser.addOption('size', help: 'Size of the triangle');
+  }
+
+  @override
+  String get name => 'equilateral';
+
+  @override
+  String get description => 'Draws an equilateral triangle';
+
+  @override
+  List<String> get aliases => ['e'];
+
+  @override
+  FutureOr<String>? run() {
+    final size = int.parse(argResults?['size'] ?? '20');
+    final char = globalResults?['char']?[0] ?? '#';
+    return drawTriangle(size, size * sqrt(3) ~/ 2, char);
+  }
+}
+
+class IsoscelesTriangleCommand extends Command<String> {
+  IsoscelesTriangleCommand() {
+    argParser.addOption('width', help: 'Width of the triangle');
+    argParser.addOption('height', help: 'Height of the triangle');
+  }
+
+  @override
+  String get name => 'isosceles';
+
+  @override
+  String get description => 'Draws an isosceles triangle';
+
+  @override
+  List<String> get aliases => ['i'];
+
+  @override
+  FutureOr<String>? run() {
+    final width = int.parse(argResults?['width'] ?? '50');
+    final height = int.parse(argResults?['height'] ?? '10');
+    final char = globalResults?['char']?[0] ?? '#';
+    return drawTriangle(width, height, char);
+  }
+}
+
+void main(List<String> args) async {
```

Comment:
Perhaps move this method to the top of the file, so it's the first thing people see? You may want to sequence the commands next, and then the support methods (`draw`, `drawTriangle`, ...).